### ISSUE TITLE
Panic crashing honeytail

### ISSUE THREAD

**Issue by ngauthier** [honeycombio/honeytail#123]
Hey folks,

We've got honeytail crashing in production. Happened yesterday but we didn't have all the logs going into honeycomb yet. Once we set up a heroku logdrain we were able to capture the panic. Here's the trace:

```
buildpack=nginx at=exit process=honeytail
	/home/travis/gopath/src/github.com/honeycombio/honeytail/leash.go:343 +0x15b
created by main.modifyEventContents.func1
	/home/travis/gopath/src/github.com/honeycombio/honeytail/leash.go:401 +0xfd6
main.modifyEventContents.func1.1(0xc4203944e0, 0xc420383800, 0xc4204e01c0, 0xc42000e108, 0xc4202eaed0, 0x8c2b40, 0xc42035e240, 0x0, 0x0, 0x0, ...)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/math/rand/rand.go:326 +0x37
math/rand.Intn(0x8000000000000000, 0xc4204080c0)
	/home/travis/.gimme/versions/go1.10.linux.amd64/src/math/rand/rand.go:166 +0x9c
math/rand.(*Rand).Intn(0xc42001e1b0, 0x8000000000000000, 0xc42035e26c)
goroutine 184 [running]:
panic: invalid argument to Intn
```

It appears to have something to do with dynamic sampling, so here's that app's dynamic sampling config:

```
AddFields = service_name=<%= ENV['HEROKU_APP_NAME'] %>-nginx
RequestShape = http.request
RequestParseQuery = all
ParserName = keyval
SampleRate = 5
DynSample = http.status
DynSample = http.upstream_cache_status
DynSample = http.request_path
```

Here's a sample of what these values look like:

```csv
Timestamp,http.request_path,http.status,http.upstream_cache_status
2018-10-27T14:13:11.991452953Z,/v1/dashboard,200,EXPIRED
2018-10-27T14:13:11.990323051Z,/v1/exchange-rates,200,STALE
2018-10-27T14:13:11.9560657Z,/v1/markets/interval,200,HIT
2018-10-27T14:13:11.937851771Z,,,
2018-10-27T14:13:11.937807612Z,,,
2018-10-27T14:13:11.928231977Z,/v1/exchange_candles,200,STALE
2018-10-27T14:13:11.875924483Z,/v1/market-cap/sparkline,200,STALE
2018-10-27T14:13:11.86590822Z,,,
2018-10-27T14:13:11.856936627Z,/v1/currencies/sparkline,200,STALE
2018-10-27T14:13:11.828181399Z,,,
2018-10-27T14:13:11.374426611Z,/v1/markets/interval,200,STALE
2018-10-27T14:13:11.263195926Z,/v1/exchange_candles,200,HIT
2018-10-27T14:13:11.176086127Z,/v1/candles,200,STALE
2018-10-27T14:13:11.162250444Z,,,
2018-10-27T14:13:11.081968323Z,/v1/prices,200,HIT
2018-10-27T14:13:10.901980656Z,,,
2018-10-27T14:13:10.901907841Z,,,
2018-10-27T14:13:10.55836332Z,/v1/market-cap/sparkline,200,HIT
2018-10-27T14:13:10.278278303Z,/v1/exchange_candles,200,MISS
2018-10-27T14:13:09.942998359Z,/v1/exchange_candles,200,EXPIRED
2018-10-27T14:13:09.822235538Z,/v1/prices,200,HIT
2018-10-27T14:13:09.613869743Z,/v1/exchange_candles,200,STALE
2018-10-27T14:13:08.692443027Z,,,
2018-10-27T14:13:08.396816014Z,/v1/market-cap/sparkline,200,HIT
2018-10-27T14:13:08.199238458Z,/v1/prices,200,EXPIRED
2018-10-27T14:13:08.191089884Z,/v1/exchange-rates,200,HIT
2018-10-27T14:13:08.189405998Z,,,
2018-10-27T14:13:08.18828458Z,/v1/markets/prices,200,EXPIRED
2018-10-27T14:13:07.980840577Z,/v1/exchange-markets/prices,200,EXPIRED
```

Also, Right before the crash I saw `http.upstream_cache_status=-` so it's worth checking if the `-` character can cause this crash as well.

Thanks and let me know if you need more information from us.

/cc @spencewood

**Comment by tredman** [comment]
Thanks for the report. I think we've figured this out (divide by zero 😱 ) - just working to get the right solution approved and then we can push a new honeytail with the fix.

**Comment by tredman** [comment]
Hey - this should be fixed now in version 1.733.

**Comment by ngauthier** [comment]
Thanks!
